### DISCUSSION TITLE
Autogenerated CRUD Resolvers

### DISCUSSION THREAD

**Discussion by coopbri** [Brendonovich/prisma-client-rust#347] [Q&A] [answered]
I know this is somewhat out of scope of PCR, but I wasn't sure where else to ask this (https://github.com/async-graphql/async-graphql maybe?). Is anyone aware of a tool in the Rust/PCR ecosystem to automatically generate GraphQL CRUD resolvers based on a Prisma schema?

Examples of tools that support this from the Node/JS/TS ecosystem:
- [GraphQL Nexus (`t.crud`)](https://nexusjs.org/docs/plugins/prisma/api#tcrud)
- [`prisma-generator-pothos-codegen`](https://github.com/Cauen/prisma-generator-pothos-codegen)

I am aware this is generally not a good practice (due to high coupling between DB schema and GraphQL API schema, making evolving each schema separately more challenging), but I love it for rapidly prototyping APIs and it can be relatively safe as long as there is granular control per entity model and the ability to compose the generated resolver logic (the aforementioned projects support these features).

P.S. PCR is awesome! Using it on some small projects and love the native rspc support.

**Comment by Brendonovich** [comment, marked as answer]
There's nothing around for PCR at the moment, but it'd definitely possible to build a custom generator that does something similar! Ideally we'd have our own GraphQL library in addition to rspc that would have tight integration with PCR, but that'll take a lot of time that we don't have atm haha.

**Reply by coopbri** [reply]
Lol totally understand, the Rust ecosystem in general is moving like a freight train. Thanks a ton for your work on Spacedrive, PCR, etc. and for answering my question :handshake:
